src/coreclr/md/enc/CMakeLists.txt

```diff
@@ -62,3 +73,16 @@ target_precompile_headers(mdruntimerw-dbi PRIVATE stdafx.h)
 add_library_clr(mdruntimerw_ppdb ${MDRUNTIMERW_SOURCES})
 target_compile_definitions(mdruntimerw_ppdb PRIVATE FEATURE_METADATA_EMIT_ALL FEATURE_METADATA_EMIT_PORTABLE_PDB)
 target_precompile_headers(mdruntimerw_ppdb PRIVATE stdafx.h)
+
+if(CLR_CMAKE_TARGET_LINUX)
+  if(FEATURE_DISTRO_AGNOSTIC_SSL)
+    target_link_libraries(mdruntimerw_wks PRIVATE System.Security.Cryptography.Native.OpenSsl-Static)
+    target_link_libraries(mdruntimerw_ppdb PRIVATE System.Security.Cryptography.Native.OpenSsl-Static)
+  else()
+    find_package(OpenSSL QUIET)
+    if(OPENSSL_FOUND)
+      target_link_libraries(mdruntimerw_wks PRIVATE OpenSSL::SSL OpenSSL::Crypto System.Security.Cryptography.Native.OpenSsl-Static)
```

Comment:
Do we need to link into the runtime build of metadata or can we link this only into the `_ppdb` variant?

If this is only going to be used by ilasm, then we should only link it into the `_ppdb` variant of the build.

Reply:
If I tweak this logic to link the OpenSSL bits into only the `_ppdb` variant, the linker complains about undefined OpenSSL symbols for ILAsm builds with non-agnostic OpenSSL. Here's what I'm trying locally:
```
  target_link_libraries(mdruntimerw_wks PRIVATE System.Security.Cryptography.Native.OpenSsl-Static)
  if(FEATURE_DISTRO_AGNOSTIC_SSL)
    target_link_libraries(mdruntimerw_ppdb PRIVATE System.Security.Cryptography.Native.OpenSsl-Static)
  else()
    find_package(OpenSSL QUIET)
    if(OPENSSL_FOUND)
      target_link_libraries(mdruntimerw_ppdb PRIVATE OpenSSL::SSL OpenSSL::Crypto System.Security.Cryptography.Native.OpenSsl-Static)
    endif(OPENSSL_FOUND)
  endif(FEATURE_DISTRO_AGNOSTIC_SSL)
```

Am I missing something?

Reply:
I agree with @jkoritzinsky that this should be linked into ilasm only. We do not want this to leak to other places by accident. It may be best to add all this under src\coreclr\ilasm.

Reply:
Got it, I've moved this under `src/coreclr/ilasm`